File: katas/6 kyu/bit-counting/solution.py
# bit-counting
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def count_bits(n):
  return sum(1 for x in list(bin(n)[2:]) if x == '1')
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def countBits(n):
  return bin(n).count("1")
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(count_bits(0), 0)
test.assert_equals(count_bits(4), 1)
test.assert_equals(count_bits(7), 3)
test.assert_equals(count_bits(9), 2)
test.assert_equals(count_bits(10), 2)

File: katas/6 kyu/bit-counting/solution.js
// bit-counting
/* ------------------------------- MY SOLUTION ------------------------------ */
var countBits = function(n) {
  return n.toString(2).split("").filter(x => x === '1').length
};
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
var countBits = function(n)
{
  return (n.toString(2).match(/1/g) || []).length;
};
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const { assert } = require("chai")

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(countBits(0), 0);
    assert.strictEqual(countBits(4), 1);
    assert.strictEqual(countBits(7), 3);
    assert.strictEqual(countBits(9), 2);
    assert.strictEqual(countBits(10), 2);
    })
  })